### PR TITLE
Adds Asthma

### PR DESCRIPTION
**Pull request by LordFowl**
Sprinting no longer deals oxygen damage unless you have lung damage or asthma. Sprinting is now based off of both halloss and oxyloss.

Lung damage now causes both oxygen damage and halloss when you sprint past your threshold.

Coughing disability now causes halloss when you sprint past your threshold.

Adds asthma, which causes oxygen damage when you sprint past your threshold, and inhibits your ability to recover from oxygen naturally. It also makes you cough harmlessly when you have more than 10 oxygen damage.

- [x] Testing.

### CODE REVIEW

**Comment by Kaedwuff** [comment]
Is this how we're getting  around the 'I ran so hard I got brain damage' issue? :O

**Comment by LordFowl** [comment]
If you consider fixing the faulty logic of somehow getting 50 oxygen damage because you ran "getting around" an issue, then yes.

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -549,7 +554,10 @@
 		failed_last_breath = 1
 	else
 		failed_last_breath = 0
-		adjustOxyLoss(-5)
+		if(disabilities & ASTHMA)
+			adjustOxyLoss(rand(-5,0))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/life.dm]: An asthmatic takes less oxygen damage while failing to breathe? Wut?

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -1627,7 +1635,8 @@
 	if (!exhaust_threshold) // Also quit if there's no exhaust threshold specified, because division by 0 is amazing.
 		return
 
-	if (failed_last_breath || oxyloss > exhaust_threshold)//Can't catch our breath if we're suffocating
+	if (failed_last_breath || (oxyloss + halloss) > exhaust_threshold)//Can't catch our breath if we're suffocating
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/life.dm]: This is going to be troublesome, perhaps. You should check what happens if you baton yourself into paincrit, can you breathe then?

File: code/modules/mob/living/carbon/human/life.dm (hunk 1 above)
Comment (LordFowl) [reply code/modules/mob/living/carbon/human/life.dm]: What indeed? This is if they pass their last_breath, hence `failed_last_breath = 0`

File: code/modules/mob/living/carbon/human/life.dm (hunk 2 above)
Comment (LordFowl) [reply code/modules/mob/living/carbon/human/life.dm]: This is only for stamina replenishment.

**Pull request merged**